From ed892103501352b1a920664d207416ca70b23d43 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sun, 10 Aug 2014 18:21:21 -0400
Subject: [iommu] vt-d: Use list_for_each_entry_safe() for dmar_domain->devices traversa

Message-id: <20140810182121.24055.96687.stgit@gir.stowe>
Patchwork-id: 87590
O-Subject: [RHEL7.1 PATCH 19/19] iommu/vt-d: Use list_for_each_entry_safe() for dmar_domain->devices traversa
Bugzilla: 1127467
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Dutile <ddutile@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1127467
Upstream Status: bca2b916f3d5c5bcb3d9d38fc1e4cb4d83b1d8a8

commit bca2b916f3d5c5bcb3d9d38fc1e4cb4d83b1d8a8
Author: Yijing Wang <wangyijing@huawei.com>
Date:   Thu Oct 31 17:26:04 2013 +0800

    iommu/vt-d: Use list_for_each_entry_safe() for dmar_domain->devices traversa

    Replace list_for_each_safe() + list_entry() with the simpler
    list_for_each_entry_safe().

    Signed-off-by: Yijing Wang <wangyijing@huawei.com>
    Signed-off-by: Joerg Roedel <joro@8bytes.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/intel-iommu.c b/drivers/iommu/intel-iommu.c
index cae5a08..59779e1 100644
--- a/drivers/iommu/intel-iommu.c
+++ b/drivers/iommu/intel-iommu.c
@@ -3781,11 +3781,10 @@ static void iommu_detach_dependent_devices(struct intel_iommu *iommu,
 static void domain_remove_one_dev_info(struct dmar_domain *domain,
        struct pci_dev *pdev)
 {
- struct device_domain_info *info;
+ struct device_domain_info *info, *tmp;
  struct intel_iommu *iommu;
  unsigned long flags;
  int found = 0;
- struct list_head *entry, *tmp;
 
  iommu = device_to_iommu(pci_domain_nr(pdev->bus), pdev->bus->number,
     pdev->devfn);
@@ -3793,8 +3792,7 @@ static void domain_remove_one_dev_info(struct dmar_domain *domain,
   return;
 
  spin_lock_irqsave(&device_domain_lock, flags);
- list_for_each_safe(entry, tmp, &domain->devices) {
-  info = list_entry(entry, struct device_domain_info, link);
+ list_for_each_entry_safe(info, tmp, &domain->devices, link) {
   if (info->segment == pci_domain_nr(pdev->bus) &&
       info->bus == pdev->bus->number &&
       info->devfn == pdev->devfn) {
-- 
1.7.1